src/update/UpdateManager.cpp

```diff
@@ -0,0 +1,150 @@
+/*!********************************************************************
+ Audacity: A Digital Audio Editor
+
+ @file UpdateManager.cpp
+ @brief Declare a class that managing of updates.
+
+ Anton Gerasimov
+ **********************************************************************/
+
+#include "UpdateManager.h"
+#include "UpdatePopupDialog.h"
+
+#include "NetworkManager.h"
+#include "IResponse.h"
+#include "Request.h"
+
+#include "widgets/AudacityMessageBox.h"
+
+#include <wx/platinfo.h>
+#include <wx/utils.h>
+
+static const char* prefsUpdatePopupDialogShown = "/Update/UpdatePopupDialogShown";
+static const char* prefsUpdateScheduledTime = "/Update/UpdateScheduledTime";
+
+enum { ID_TIMER = wxID_HIGHEST + 1 };
+
+BEGIN_EVENT_TABLE(UpdateManager, wxEvtHandler)
+    EVT_TIMER(ID_TIMER, UpdateManager::OnTimer)
+END_EVENT_TABLE()
+
+UpdateManager::UpdateManager(AudacityProject& project)
+    : mTrackingInterval(
+        std::chrono::milliseconds(std::chrono::hours(12)).count())
+{
+    mParent = (wxWindow*)(&GetProjectFrame(project));
+    wxASSERT(mParent);
+
+    mTimer.SetOwner(this, ID_TIMER);
+    mTimer.StartOnce();
+}
+
+UpdateManager::~UpdateManager()
+{
+    mTimer.Stop();
+}
+
+void UpdateManager::enableTracking(bool enable)
```

Comment:
We should probably rename `tracking` to `UpdatesChecking` or whatever. Tracking has a lot of negativity attached to it.